### PR TITLE
Fixed screen map updates for invisible mobile actors

### PR DESCRIPTION
**Pull request by Mailaender**
Some traits already kept that in mind, but not all of them. Fixes https://github.com/OpenRA/OpenRA/issues/10766 and related crashes.

### CODE REVIEW

**Comment by RoosterDragon** [comment]
My bad for missing these originally - perhaps we should factor out some helpers methods for adds, updates and removes to fix this more fully.

**Comment by Mailaender** [comment]
You mean we should do the boundary checks in https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Game/Traits/World/ScreenMap.cs#L78?

**Comment by RoosterDragon** [comment]
I was thinking of methods just to reduce the duplication of:
```c#
self.World.ActorMap.AddInfluence(self, this);
self.World.ActorMap.AddPosition(self, this);
self.World.ScreenMap.Add(self);
```

but moving the checks to ScreenMap is also a good idea and we should probably do that. Why not both?

**Comment by Mailaender** [comment]
I summarized them in a new function. This will now also check everything against `IsInWorld` before doing actor influence changes.

File: OpenRA.Game/World.cs [hunk 1]
```diff
@@ -193,6 +193,35 @@ internal World(Map map, OrderManager orderManager, WorldType type)
...
+			ActorMap.UpdatePosition(self, ios);
+		}
+
+		public void RemoveFromMaps(Actor self, IOccupySpace ios)
+		{
+			ActorMap.RemoveInfluence(self, ios);
+			ActorMap.RemovePosition(self, ios);
+
+			if (!self.Bounds.Size.IsEmpty)
+				self.World.ScreenMap.Remove(self);
```
Comment (RoosterDragon) [inline OpenRA.Game/World.cs]: Maybe just `ScreenMap.Remove(self);` for consistency.

**Comment by RoosterDragon** [comment]
:+1:

File: OpenRA.Game/World.cs (hunk 1 above)
Comment (Mailaender) [inline OpenRA.Game/World.cs]: Woops. Fixed.

**Comment by obrakmann** [comment]
:+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/3aabb4e3b210d335773e1d26056c212c3bcf60c8)

**Pull request merged**